lib/phoenix/live_dashboard/components/tab_bar_component.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Phoenix.LiveDashboard.TabBarComponent do
+  @moduledoc false
+  use Phoenix.LiveDashboard.Web, :live_component
+
+  @impl true
+  def mount(socket) do
+    {:ok, socket}
+  end
+
+  @impl true
+  def update(assigns, socket) do
+    %{page: page, entries: entries} = assigns
```

Comment:
Now that the component is called `tab_bar`, what do you think about renaming `entries` to `tabs`?

Also, we should probably validate that all tabs have the `render` option. We could do this here but the stracktrace can be confusing... what if we do so when `tab_bar/1` is invoked? We should probably do the same for tables as well, do all validation on the `table` call and leave `update` only to update the state.

Reply:
Renamed to `tabs`!

I created the first draft of the param validations of the TabBarComponent. Maybe I validated too much :P and I have to improve the error messages.

In general, I see it makes a lot of sense. In order to compose different components, it will make easier to understand the parameter errors. The component code will be easier to understand as well, splitting updating and validation into two different functions.

Reply:
Another option is to use the NimbleOptions to validate component parameters. It is a nice and small library, maybe you know it? :stuck_out_tongue: 
However, it would be an additional dependency and it's not worth it